cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/common/DBUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.common;
+
+import co.cask.cdap.template.etl.batch.sink.DBSink;
+import co.cask.cdap.template.etl.batch.source.DBSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Utility methods for Database plugins shared by {@link DBSource} and {@link DBSink}
+ */
+public final class DBUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(DBUtils.class);
+
+  /**
+   * Performs any Database related cleanup
+   *
+   * @param driverClass the JDBC driver class
+   */
+  public static void cleanup(Class<? extends Driver> driverClass) {
+    shutDownMySQLAbandonedConnectionCleanupThread(driverClass.getClassLoader());
+  }
+
+  /**
+   * De-register all SQL drivers that are associated with the class
+   */
+  public static void deRegisterDriver(Class<? extends Driver> driverClass)
+                     throws NoSuchFieldException, IllegalAccessException, ClassNotFoundException {
+    Field field = DriverManager.class.getDeclaredField("registeredDrivers");
+    field.setAccessible(true);
+    CopyOnWriteArrayList<?> list = (CopyOnWriteArrayList<?>) field.get(null);
```

Comment:
@chtyim ``DriverManager.getDrivers()`` internally has a filter to only return Drivers that were added with the caller's classloader. It also doesn't expose a method with a classloader parameter. Hence we need to use reflection here.